src/plugins/workspace/server/workspace_client.ts

```diff
@@ -102,6 +106,13 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
       if (existingWorkspaceRes && existingWorkspaceRes.total > 0) {
         throw new Error(DUPLICATE_WORKSPACE_NAME_ERROR);
       }
+
+      if (dataSources) {
+        for (const ds of dataSources) {
+          await client.addToWorkspaces(DATA_SOURCE_SAVED_OBJECT_TYPE, ds.id, [id]);
```

Comment:
Instead of calling `client.addToWorkspaces` sequentially, what about doing something like `Promise.all`?

Reply:
Updated.